File: runtime/csharp/Prompty.Foundry/FoundryProcessor.cs
// Copyright (c) Microsoft. All rights reserved.

namespace Prompty.Foundry;

/// <summary>
/// Processes Azure OpenAI / Foundry responses.
/// Delegates entirely to OpenAIProcessor — same SDK, same response types.
/// Registered under keys "foundry" and "azure" (deprecated alias).
/// </summary>
public class FoundryProcessor : OpenAI.OpenAIProcessor
{
}

File: runtime/java/prompty-foundry/src/main/java/com/microsoft/prompty/foundry/FoundryProcessor.java
package com.microsoft.prompty.foundry;

import com.microsoft.prompty.openai.OpenAIProcessor;

/**
 * Reads Azure OpenAI and Foundry responses.
 *
 * <p>Azure returns OpenAI's response shape, so all of the reading is inherited. Only the provider
 * name changes, and the fact that none of Azure's endpoints hand back a handle that would let a
 * later request resume model-visible state — so the conversation is retained as explicitly portable
 * context instead of a continuation the provider could not honour.
 */
public class FoundryProcessor extends OpenAIProcessor {

  @Override
  protected String providerName() {
    return "foundry";
  }

  @Override
  protected boolean supportsResponsesContinuation() {
    return false;
  }
}
